engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -176,12 +195,25 @@ class PeriodicProcessService(delegateDeploymentManager: DeploymentManager,
     } yield handleEvent(FinishedEvent(currentState, state))
   }
 
+  private def markFailedOnDeployAction(deployment: PeriodicProcessDeployment, state: Option[ProcessState]): RepositoryAction[Unit] = {
+    val retryAt =
+      if (deployment.retries < deployMaxRetries)
+        Some(now().plus(deployRetryPenalize.toMillis, ChronoUnit.MILLIS))
+      else
+        deployment.retryAt
+    logger.info(s"Marking ${deployment.display} as failed on deploy. Retry no: ${deployment.retries}")
+    for {
+      _ <- scheduledProcessesRepository.markFailedOnDeploy(deployment.id, deployment.retries, retryAt)
+      currentState <- scheduledProcessesRepository.findProcessData(deployment.id)
+    } yield handleEvent(FailedOnDeployEvent(currentState, state))
+  }
+
   private def markFailedAction(deployment: PeriodicProcessDeployment, state: Option[ProcessState]): RepositoryAction[Unit] = {
-    logger.info(s"Marking ${deployment.display} as failed")
+    logger.info(s"Marking ${deployment.display} as failed.")
     for {
       _ <- scheduledProcessesRepository.markFailed(deployment.id)
       currentState <- scheduledProcessesRepository.findProcessData(deployment.id)
-    } yield handleEvent(FailedEvent(currentState, state))
+    } yield handleEvent(FailedOnRunEvent(currentState, state))
```

Comment:
What about retries/retry_at? Why we don't handle it here? In findToBeRetried we increase retries counter (but not date), but increasing it there is a bit unclear (as it's in find... method)?

Reply:
:+1: I agree,

Reply:
done